src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -296,11 +296,13 @@ def validate_source_uri(namespace):  # pylint: disable=too-many-statements
     if snapshot:
         query_params.append('snapshot={}'.format(snapshot))
 
+    from six.moves.urllib.parse import quote  # pylint: disable=import-error
+
     uri = 'https://{0}.{1}.{6}/{2}/{3}{4}{5}'.format(
         source_account_name,
         'blob' if valid_blob_source else 'file',
         container if valid_blob_source else share,
-        blob if valid_blob_source else path,
+        quote(blob if valid_blob_source else path, '/()$=\',~'),
```

Comment:
Why don't you use the util function here?

Reply:
I forgot 
...